task: VolikIA/Sale-17

volik25 opened the pull request:
Дерево директорий + добавление новой директории

volik25 committed:
task: VolikIA/Sale-17
Дерево директорий + добавление новой директории

ianomokonov commented on src/app/_services/back/script.service.ts:
```diff
@@ -2,15 +2,24 @@ import { HttpClient } from '@angular/common/http';
+    return this.http.get<Script>(`${this.baseUrl}/script/${id}`);
+  }
+
+  public getScripts(filters?: any): Observable<Script[]> {
```
Договорились вынести получение папок в отдельный метод getFolders
Параметр filters можно оставить для дальнейшей реализации фильтров

ianomokonov commented on src/app/_services/back/script.service.ts:
```diff
@@ -2,15 +2,24 @@ import { HttpClient } from '@angular/common/http';
+  }
+
+  public getScripts(filters?: any): Observable<Script[]> {
+    return this.http.get<Script[]>(`${this.baseUrl}/script`, { params: filters });
```
строка должна быть ${this.baseUrl}/scripts
потому что получаешь список

ianomokonov committed:
Merge branch 'master' into feature/SALE-17

ianomokonov committed:
SALE-21: бэк для создания и получения папок и скриптов

ianomokonov committed:
SALE-17: поправил вход и регистрацию

ianomokonov committed:
Merge branch 'master' into feature/SALE-17

ianomokonov committed:
SALE-17: redirect to scripts

volik25 committed:
Sale-17
Доавлена модалка добавления скрипта
Фикс лэйблов при входе и регистрации
Реализован клиент-сервер в разделе скриптов

ianomokonov commented on src/app/modules/personal-cabinet/_modals/add-script/add-script.component.ts:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  ngOnInit(): void {
+    this.scriptService.getFolders().subscribe((folders) => {
```
Нужно перенести загрузку в родительский компонент и передавать сюда в конфиге, чтоб не грузить при каждом открытии модалки

В создании папки также

ianomokonov commented on src/app/modules/personal-cabinet/_modals/add-script/add-script.component.ts:
```diff
@@ -0,0 +1,43 @@
+    private fb: FormBuilder,
+    private modal: DynamicDialogRef,
+  ) {
+    this.scriptForm = fb.group({
```
у нас получается два одинаковых компонента? думаю нужно объединить в один
Два пути вижу, либо в конфиг передавать флаг isFolder и на основе него выбиарть какой запрос отправлять, либо добавить на форму switch (писал в задаче https://trello.com/c/5Bggn19p/16-%D1%81%D0%BE%D0%B7%D0%B4%D0%B0%D0%BD%D0%B8%D1%8F-%D1%81%D0%BA%D1%80%D0%B8%D0%BF%D1%82%D0%B0) типо что создаем, папку или скрипт

Думаю первый вариант лучше

ianomokonov commented on src/app/components/sign-in/sign-in.component.html:
```diff
@@ -8,6 +8,7 @@ <h3 class='col-6 m-auto text-center'>Авторизация</h3>
       <span class="p-float-label p-input-icon-left">
           <i class="pi pi-user"></i>
           <input pInputText
+                 [class.p-filled]='logInForm.controls.login.value'
```
такое на работает, думаю нужно через ngClass написать, потому что в таком синтаксисе дефисы не читаются

![image](https://user-images.githubusercontent.com/25670622/132071398-e6cb473e-2a75-41d3-80c4-57dea6d37f42.png)

ianomokonov replied:
Это касается всех полей

volik25 replied:
Наоборот через ngClass не работает, а в таком формате всё по кайфу

volik25 replied:
У тебя что-то не то

volik25 replied:
![image](https://user-images.githubusercontent.com/45817026/132095890-673500d4-b47d-4549-aadb-377579b095fa.png)

volik25 replied:
Либо же автокомплит - это не запись в value контрола и его надо нахуй запретить

ianomokonov replied:
Поменяй лейблы тогда, баг не исправлен

volik25 committed:
Sale-17
Убран лишний запрос на папки в модальном окне

volik25 committed:
Sale-17
Одно модальное окно с флагом

volik25 committed:
Sale-17
Фикс формы входа
